test-result-summary-client/src/PerfCompare/PerfCompare.jsx

```diff
@@ -489,41 +484,76 @@ export default class PerfCompare extends Component {
+                    curTestCI = parsedVariantsTest.metrics[j].value.CI;
                 } else {
                     // perffarm builds
                     let runBaseline = this.state.benchmarkRuns.benchmarkRunBaseline;
                     let runTest = this.state.benchmarkRuns.benchmarkRunTest
                     curRawValues = {
-                        "baseline": runBaseline.parsedVariants[i].metrics[j].value,
-                        "test": runTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].value
+                        "baseline": Object.entries(runBaseline.parsedVariants[i].metrics[j]),
+                        "test": Object.entries(runTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex])
                     };
+                    curBaselineScore = parsedVariantsBaseline.metrics[j]['mean'];
+                    curBaselineCI = parsedVariantsBaseline.metrics[j]['CI']/100;
+                    curTestScore = parsedVariantsTest.metrics[j]['mean'];
+                    curTestCI = parsedVariantsTest.metrics[j]['CI']/100;
                 }
 
-                curBaselineScore = parsedVariantsBaseline.metrics[j].value.mean;
-                curBaselineCI = parsedVariantsBaseline.metrics[j].value.CI;
-                curTestScore = parsedVariantsTest.metrics[j].value.mean;
-                curTestCI = parsedVariantsTest.metrics[j].value.CI;
-
                 // Get the metric's units
+                //Parse benchmark Variant and verify if the infos there
+                //get BenchmarkRouter & Metric files from server
+                let benchmarkMetricRequest = await fetch( `/api/getBenchmarkVariants`, {
+                    method: 'get'
+                } );
+                let benchmarkMetricVariants = await benchmarkMetricRequest.json();
+                var BenchmarkRouter = benchmarkMetricVariants['benchmarkMetricRouter'];
+                var BenchmarkMetric = benchmarkMetricVariants['benchmarkMetric'];
+                let isValid = true;
+                if (!BenchmarkRouter[curVariantData["benchmark"]]) {
+                    isValid = false;
+                } else if (!BenchmarkRouter[curVariantData["benchmark"]][curVariantData["variant"]]) {
+                    isValid=false;
+                } else if (!BenchmarkMetric[BenchmarkRouter[curVariantData["benchmark"]][curVariantData["variant"]]]
+                || !BenchmarkMetric[BenchmarkRouter[curVariantData["benchmark"]][curVariantData["variant"]]]["metrics"]
+                || Object.keys(BenchmarkMetric[BenchmarkRouter[curVariantData["benchmark"]][curVariantData["variant"]]]["metrics"]).length == 0) {
+                    isValid = false;
+                } else {
+                    curBenchmarkVariant = BenchmarkMetric[BenchmarkRouter[curVariantData["benchmark"]][curVariantData["variant"]]]["metrics"];
+                }
```

Comment:
Don't think we need this. We could just put it under `try` below. No need for `isValid` either.